src/test/java/io/stargate/sgv2/jsonapi/api/v1/CreateCollectionIntegrationTest.java

```diff
@@ -47,6 +48,152 @@ public void happyPath() {
+              "options" : {
+                "vector" : {
+                "size" : 6,
+                  "function" : "cosine"
+                }
+              }
+            }
+          }
+          """;
+      // create a non vector collection
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(createNonVectorCollectionJson)
+          .when()
+          .post(NamespaceResource.BASE_PATH, namespaceName)
+          .then()
+          .statusCode(200)
+          .body("status.ok", is(1));
+
+      // recreate the same non vector collection
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(createNonVectorCollectionJson)
+          .when()
+          .post(NamespaceResource.BASE_PATH, namespaceName)
+          .then()
+          .statusCode(200)
+          .body("status.ok", is(1));
+
+      // create a vector collection with the same name
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(createVectorCollection)
+          .when()
+          .post(NamespaceResource.BASE_PATH, namespaceName)
+          .then()
+          .body("status", is(nullValue()))
+          .body("data", is(nullValue()))
+          .body(
+              "errors[0].message",
+              is("The provided collection already exists with a non-vector setting"))
+          .body("errors[0].errorCode", is("INVALID_COLLECTION_NAME"))
+          .body("errors[0].exceptionClass", is("JsonApiException"));
+
+      // delete the collection
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(deleteCollectionJson.formatted("simple_collection"))
+          .when()
+          .post(NamespaceResource.BASE_PATH, namespaceName)
+          .then()
+          .statusCode(200)
+          .body("status.ok", is(1));
+
+      // create a vector collection
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(createVectorCollection)
+          .when()
+          .post(NamespaceResource.BASE_PATH, namespaceName)
+          .then()
+          .statusCode(200)
+          .body("status.ok", is(1));
+
+      // recreate the same vector collection
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(createVectorCollection)
+          .when()
+          .post(NamespaceResource.BASE_PATH, namespaceName)
+          .then()
+          .statusCode(200)
+          .body("status.ok", is(1));
+
+      // create a non vector collection with the same name
```

Comment:
This seems a bit strange: I would expect `createCollection()` would fail if calling `createCollection()` with no options, but there already is a collection with different options.

FWIW, MongoDB always fails if you call `createCollection()` on a collection that already exists, even if with same option. Below is a Mongoose example:

```
'use strict';

const mongoose = require('mongoose');

run().catch(err => console.log(err));

async function run() {
  await mongoose.connect('mongodb://127.0.0.1:27017/mongoose_test');

  await mongoose.connection.dropCollection('foobar').catch(() => {});
  await mongoose.connection.createCollection('foobar', { capped: true, size: 1024 });
  // Throws "Collection already exists. NS: mongoose_test.foobar"
  await mongoose.connection.createCollection('foobar', {});
  console.log('Done', await mongoose.connection.db.listCollections().toArray());
}
```

Reply:
I think this is not what user would expect -- if they want non-vector Collection, service should not quietly just leave existing vector Collection in place and proceed.
It should indicate an error.

Reply:
I think we should error if the configuration are different. It doesn't matter if existing collection is vector enabled or not. Also this will be helpful for future enhancement like upsert support for vector collection. 
If a collection is created for vector and user want to use it as non vector collection, they can still use it but configuration can't be changed.

Reply:
Got it, thank you for pointing it out! Solved now!